pdfminer/layout.py

```diff
@@ -416,13 +416,43 @@ def add(self, obj):
         return
 
     def find_neighbors(self, plane, ratio):
+        """
+        Finds neighboring LTTextLineHorizontals in the plane.
+
+        Returns a list of other LTTestLineHorizontals in the plane which are
+        close to self. "Close" can be controlled by ratio. The returned objects
+        will be the same height as self, and also either left-, right-, or
+        centrally-aligned.
+        """
         d = ratio*self.height
         objs = plane.find((self.x0, self.y0-d, self.x1, self.y1+d))
         return [obj for obj in objs
                 if (isinstance(obj, LTTextLineHorizontal) and
-                    abs(obj.height-self.height) < d and
-                    (abs(obj.x0-self.x0) < d or
-                     abs(obj.x1-self.x1) < d))]
+                    self.__is_same_height_as(obj, tolerance=d) and
+                    (self.__is_left_aligned_with(obj, tolerance=d) or
+                     self.__is_right_aligned_with(obj, tolerance=d) or
+                     self.__is_centrally_aligned_with(obj, tolerance=d)))]
+
+    def __is_left_aligned_with(self, other, tolerance=0):
+        """
+        Whether the left-hand edge of `other` is within `tolerance`.
+        """
+        return abs(other.x0 - self.x0) <= tolerance
+
+    def __is_right_aligned_with(self, other, tolerance=0):
+        """
+        Whether the right-hand edge of `other` is within `tolerance`.
+        """
+        return abs(other.x1 - self.x1) <= tolerance
+
+    def __is_centrally_aligned_with(self, other, tolerance=0):
+        """
+        Whether the horizontal center of `other` is within `tolerance`.
+        """
+        return abs((other.x0+other.x1)/2 - (self.x0+self.x1)/2) <= tolerance
```

Comment:
Add some spaces around the `+` and `/` signs

Reply:
Done in new commit "Cosmetic changes from code review"